hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/record/ExternalizedRecordCustomizer.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.workflows.record;
+
+import com.hedera.hapi.node.base.Transaction;
+import java.util.function.UnaryOperator;
+
+/**
+ * Allows a {@link com.hedera.node.app.spi.workflows.TransactionHandler} that dispatches child transactions
+ * to customize exactly how these records are externalized. Specifically, it allows the handler to,
+ * <ul>
+ *     <li>Completely suppress the record because it contains redundant information (as in the case of
+ *     the child transaction dispatched to implement a top-level HAPI {@code ContractCreate}).</li>
+ *     <li>Transform the dispatched {@link Transaction} immediately before it is externalized (as
+ *     in the case of the child {@link com.hedera.hapi.node.token.CryptoCreateTransactionBody} dispatched
+ *     to implement an internal contract creation, which should be externalized as an equivalent
+ *     {@link com.hedera.hapi.node.contract.ContractCreateTransactionBody}.</li>
+ * </ul>
+ */
+@FunctionalInterface
+public interface ExternalizedRecordCustomizer extends UnaryOperator<Transaction> {
+    ExternalizedRecordCustomizer NOOP_EXTERNALIZED_RECORD_CUSTOMIZER = tx -> tx;
```

Comment:
You know, why bother to define `UnaryOperator::identity` when this obvious idiom `i -> i` is so much shorter and at least as clear to read? 

(Plus I couldn't even figure out how to use it here as a method reference due to language/type system limitations.)